modules/sievefilters/functions.php

```diff
@@ -284,6 +284,18 @@ function get_sieve_client_factory($site_config)
     }
 }
 
+if (!hm_exists('decode_blocked_senders_script')) {
+    function decode_blocked_senders_script($script)
```

Comment:
This is nice, thanks! However, I still see 6 occurrences of base64_decode in sievefilters/modules.php - they should all be abstracted. Try to think about common operations that are repeated and abstract them in functions. Even better would be to have a meaningful functions doing one action at a time and calling others when necessary. I see 4 operations that are repeatedly doing the same thing - getting a script from cypht, decoding it, encoding the contents back and saving it back to cypht.